### PR TITLE
Dispatch actions when the user interact with the app

### PR DESCRIPTION
**Pull request by AlfredoElizarraras**
#### This pull request includes the following:
- [x] Dispatch addBook action from BookForm component
- [x] Dispatch removeBook action from BookList component to Book component

### CODE REVIEW

File: src/components/BookForm/BookForm.js
Comment (shloch) [inline src/components/BookForm/BookForm.js]: this could be done in one function and according to the requirements, I think it was asked that this function be named `'handleChange'`:
```
handleChange({ target: { name, value } }) {
  this.setState({
    [name]: value,
  })
}
```

File: src/components/BookForm/BookForm.js
Comment (shloch) [inline src/components/BookForm/BookForm.js]: this function should normally take an `event` as parameter, and prevent default form-submission javascript behavior: `event.preventDefault();`:

### see code below for comprehension

```
handleSubmit = (event) => {
            
            event.preventDefault();
            const { title, category } = this.state;
    	    const { addBook } = this.props;
            const book = {
               id: uuid(),
               title,
               category,
	    };
           addBook(book);

          this.setState({ title: '', category: BOOK_CATEGORIES[0] });
  }

```

Also, it's adviceable to bind these functions (`updateCategory` + `handleSubmit`) to the`this` ( learn about that here --> `https://www.youtube.com/watch?v=kVWpBtRjkCk` )

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @AlfredoElizarraras 

You did a wonderful job. There's however very little things to polish before final thumbs up, but I think you know what you are doing and therefore is on the right track. I just dropped some hints for you to get inspiration from and polish this milestone.

Waiting for a last review hopefully.

Happy coding

**Comment by AlfredoElizarraras** [comment]
### Hi @shloch :wave:

#### Thanks for your review :pray: 
#### I will apply the required changes and then submit again for another review :raised_hands:

**Review by uche-inyama — APPROVED** [review]
## Approved

### Hi Alfred :wave: 

Go on with your next task.


### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

**Comment by AlfredoElizarraras** [comment]
## Hi @uche-inyama :wave: 
### Thanks for your review and your approval :pray: 
### Have a great day too :smiley:

**Pull request merged**
